### ISSUE TITLE
indefinite promises

### ISSUE THREAD

**Issue by basaran** [timgit/pg-boss#288]
Hello again :)

A line of code is said to be worth a thousand words, so could you please watch the movie below and let me know if you think this might be an issue with pg-boss:

```js
// in an unknown Class extends EventEmitter. constructor...
        this.resolve;
        this.reject;
        this.errors = 0;
        this.swear = new Promise((resolve, reject) => {
            this.resolve = resolve;
            this.reject = reject;
            setInterval(() => {}, 10000);
        });

        this.swear.resolve = this.resolve;
        this.swear.reject = this.reject;

        this.on("stop", (x) => {
            console.log("stop received");
            this.swear.reject();
        });

       return this;
```

```js
// in an unknown subscription handler

async function longTask() {
    const task = new Task();
    task.boot("/");
    await task.swear;
}
```

```js
// back in the unknown Class extending eventEmitter boot fn...
async boot() {
  try {
     const apiResults = await fetchSomeApi();
    
     if (this.errors >= 10) return;
   } catch (e) {
    this.emit("stop", e);
 }
}
```

I'm basically creating an indefinite promise, for pg-boss to hold on to ,and rejecting it  later if need be. Hope this makes sense. The dumb setInterval is there to keep the promise open, otherwise node I suppose claims it or something.

**Comment by basaran** [comment]
It appears to work fine, just asking if you think this might blow up in my face later which happens more often than it should :)

**Comment by timgit** [comment]
pg-boss will simply wait on the promise. There are no more rules. :)

**Comment by basaran** [comment]
arigato gosaymasu, closing again.
